### DISCUSSION TITLE
Calling variables from .env file

### DISCUSSION THREAD

**Discussion by vladparl** [pactumjs/pactum#211] [Q&A] [answered]
Hello, I need to use different variables for different environments. I have this solution which is working, but is not suitable for more different variables:

**Variables are saved in .env file**
```env
APP_USERNAME = value
APP_PASSWORD = value
```

**Step is defined in steps.js file**
```js
When('login with valid credentials', {timeout: 30000}, function () {
  spec = pactum.spec();
  spec.post("/auth/login");
  spec.withJson({
    username: process.env.APP_USERNAME,
    password: process.env.APP_PASSWORD
    })
});
```

**Call the step in test.feature file**
When login with valid credentials

But, I need a way to call .env variables directly from test.feature file. Is there any way to do that? 
My point is to use same step defined in steps.js, but change the variables in feature file.

**Comment by ASaiAnudeep** [comment, marked as answer]
Update your step definition to accept custom username and password

```js
When(/^login with (.*) and (.*)$/, {timeout: 30000}, function (username, password) {
  spec = pactum.spec();
  spec.post("/auth/login");
  spec.withJson({
    username: process.env[username],
    password: process.env[password]
    })
});
```

**Reply by vladparl** [reply]
Thank you, works like a charm!
